File: DAVID.py

tax = 0
def applyTax(state):
    if state=="Alabama":
        tax = 1.04
    elif state=="Alaska":
        tax = 1
    elif state=="Arizona":
        tax = 1.056
    elif state=="Arkansas":
        tax = 1.065
    elif state=="California":
        tax = 1.0725
    elif state=="Colorado":
        tax = 1.029
    elif state=="Conneticut":
        tax = 1.0635
    elif state=="Delaware":
        tax = 1
    elif state=="District of Columbia":
        tax = 1.0575
    elif state=="Florida":
        tax = 1.06
    elif state=="Georgia":
        tax = 1.04
    elif state=="Hawaii":
        tax = 1.04
    elif state=="Idaho":
        tax = 1.06
    elif state=="Illinois":
        tax = 1.0625
    elif state=="Indiana":
        tax = 1.07
    elif state=="Iowa":
        tax = 1.06
    elif state=="Kansas":
        tax = 1.065
    elif state=="Kentucky":
        tax = 1.06
    elif state=="Louisiana":
        tax = 1.0445
    elif state=="Maine":
        tax = 1.055
    elif state=="Maryland":
        tax = 1.06
    elif state=="Massachusetts":
        tax = 1.0625
    elif state=="Michigan":
        tax = 1.06
    elif state=="Minnesota":
        tax = 1.0688
    elif state=="Mississippi":
        tax = 1.07
    elif state=="Missouri":
        tax = 1.0423
    elif state=="Montana":
        tax = 1
    elif state=="Nebraska":
        tax = 1.055
    elif state=="Nevada":
        tax = 1.046
    elif state=="New Hampshire":
        tax = 1
    elif state=="New Jersey":
        tax = 1.0663
    elif state=="New Mexico":
        tax = 1.0513
    elif state=="New York":
        tax = 1.04
    elif state=="North Carolina":
        tax = 1.0475
    elif state=="North Dakota":
        tax = 1.05
    elif state=="Ohio":
        tax = 1.0575
    elif state=="Oklahoma":
        tax = 1.045
    elif state=="Oregon":
        tax = 1
    elif state=="Pennsylvania":
        tax = 1.06
    elif state=="Puerto Rico":
        tax = 1.06
    elif state=="Rhode Island":
        tax = 1.07
    elif state=="South Carolina":
        tax = 1.06
    elif state=="South Dakota":
        tax = 1.045
    elif state=="Tennessee":
        tax = 1.07
    elif state=="Texas":
        tax = 1.0625
    elif state=="Utah":
        tax = 1.047
    elif state=="Vermont":
        tax = 1.06
    elif state=="Virginia":
        tax = 1.043
    elif state=="Washington":
        tax = 1.065
    elif state=="Washington, D.C.":
        tax = 1.06
    elif state=="West Virginia":
        tax = 1.06
    elif state=="Wisconsin":
        tax = 1.05
    elif state=="Wyoming":
        tax = 1.05

def priceCalc(base,tax):
    applyTax(state)
    return round(base*tax,2)

File: DAVID.js

let tax = 0;

function applyTax(state) {
    const taxRates = {
        "Alabama": 1.04,
        "Alaska": 1.00,
        "Arizona": 1.056,
        "Arkansas": 1.065,
        "California": 1.0725,
        "Colorado": 1.029,
        "Connecticut": 1.0635,
        "Delaware": 1.00,
        "District of Columbia": 1.0575,
        "Florida": 1.06,
        "Georgia": 1.04,
        "Hawaii": 1.04,
        "Idaho": 1.06,
        "Illinois": 1.0625,
        "Indiana": 1.07,
        "Iowa": 1.06,
        "Kansas": 1.065,
        "Kentucky": 1.06,
        "Louisiana": 1.0445,
        "Maine": 1.055,
        "Maryland": 1.06,
        "Massachusetts": 1.0625,
        "Michigan": 1.06,
        "Minnesota": 1.0688,
        "Mississippi": 1.07,
        "Missouri": 1.0423,
        "Montana": 1.00,
        "Nebraska": 1.055,
        "Nevada": 1.046,
        "New Hampshire": 1.00,
        "New Jersey": 1.0663,
        "New Mexico": 1.0513,
        "New York": 1.04,
        "North Carolina": 1.0475,
        "North Dakota": 1.05,
        "Ohio": 1.0575,
        "Oklahoma": 1.045,
        "Oregon": 1.00,
        "Pennsylvania": 1.06,
        "Puerto Rico": 1.06,
        "Rhode Island": 1.07,
        "South Carolina": 1.06,
        "South Dakota": 1.045,
        "Tennessee": 1.07,
        "Texas": 1.0625,
        "Utah": 1.047,
        "Vermont": 1.06,
        "Virginia": 1.043,
        "Washington": 1.065,
        "Washington, D.C.": 1.06,
        "West Virginia": 1.06,
        "Wisconsin": 1.05,
        "Wyoming": 1.05
    };

    tax = taxRates[state] || 1.00;  // Default to 1.00 if state is not found
}

function priceCalc(base, state) {
    applyTax(state);
    return Math.round(base * tax * 100) / 100;
}

